core/src/main/java/hudson/model/Cause.java

```diff
@@ -436,7 +455,7 @@ public String getShortDescription() {
         public void print(TaskListener listener) {
             listener.getLogger().println(Messages.Cause_UserIdCause_ShortDescription(
                     // TODO better to use ModelHyperlinkNote.encodeTo(User), or User.getUrl, since it handles URL escaping
-                    ModelHyperlinkNote.encodeTo("/user/"+getUserId(), getUserName())));
+                    ModelHyperlinkNote.encodeTo("/user/"+getUserIdOrUnknown(), getUserName())));
```

Comment:
Shouldn't this use `getUserUrl` to begin with?

Reply:
As in my comment above?

Reply:
I just decided not to tough it since it does not make the thing worse